### PR TITLE
IDE-373 2 level KEL imports not working

### PR DESCRIPTION
**Pull request by dehilsterlexis**
Fixed IDE-373

### CODE REVIEW

**Comment by dehilsterlexis** [comment]
@GordonSmith for review

File: comms/Attribute.cpp [hunk 1]
```diff
@@ -487,9 +487,9 @@ class CAttribute : public CAttributeBase, public IAttribute, public IAttributeHi
     }
 #endif
 
-    int PreProcess(PREPROCESS_TYPE action, const TCHAR * overrideEcl, IAttributeVector & attrs, Dali::CEclExceptionVector & errs) const
+    int PreProcess(PREPROCESS_TYPE action, const TCHAR * overrideEcl, IAttributeVector & attrs, std::map<std::pair<PREPROCESS_TYPE, std::_tstring>, bool> & attrProcessed, Dali::CEclExceptionVector & errs) const
```
Comment (GordonSmith) [inline comms/Attribute.cpp]: Minor:  Add a typedef for this declaration (similar to wht we do for IAttributeVector)

File: comms/AttributeImpl.h [hunk 2]
```diff
@@ -66,8 +66,15 @@ class CAttributeBase : public clib::CLockableUnknown
         return false;
     }
 
-    virtual int PreProcess(PREPROCESS_TYPE action, const TCHAR * overrideEcl, IAttributeVector & attrs, Dali::CEclExceptionVector & errs) const
+    virtual int PreProcess(PREPROCESS_TYPE action, const TCHAR * overrideEcl, IAttributeVector & attrs, std::map<std::pair<PREPROCESS_TYPE, std::_tstring>, bool> & attrProcessed, Dali::CEclExceptionVector & errs) const
     {
+        std::_tstring label = _T("");
+        label = (boost::_tformat(_T("%1%.%2%")) % GetModuleQualifiedLabel(true) % GetLabel()).str();
```
Comment (GordonSmith) [inline comms/AttributeImpl.h]: Combine with previous line.

File: comms/AttributeImpl.h [hunk 3]
```diff
@@ -146,9 +154,11 @@ class CAttributeBase : public clib::CLockableUnknown
 
             errorFile.Create(NULL, GENERIC_READ);
             errorFile.HandsOff();
-            
-            std::_tstring cmd = (boost::_tformat(_T("cmd /c %1% %2% %3% %4% %5% %6% %7% %8% \"%9%\"")) % 
-                batchFile.c_str() % PREPROCESS_LABEL[action] % GetModuleQualifiedLabel(true) % GetLabel() % 
+
+            label = (boost::_tformat(_T("%1%.%2%")) % GetModuleQualifiedLabel(true) % GetLabel()).str();
```
Comment (GordonSmith) [inline comms/AttributeImpl.h]: Not used here?

**Comment by dehilsterlexis** [comment]
@GordonSmith back to you

**Comment by dehilsterlexis** [comment]
Eric and Shawn found a bug with multiple include files in one file. I have fixed it and pushed it and they are testing it now. Will let you know if it did the trick.

**Comment by dehilsterlexis** [comment]
@GordonSmith for review

**Comment by richardkchapman** [comment]
Should this be targetting 6.4.0? 6.2.12?

**Comment by dehilsterlexis** [comment]
I think we should keep it a 6.4. I have some bug fixes after this that need to be included and also we have some of the batch file stuff we should get in as well. That would be a more complete set to release all at once and it will be better tested.

**Comment by richardkchapman** [comment]
At present it is targetting 7.0 ... To target 6.4.0 will require you to rebase --onto candidate-6.4.0 HEAD^, repush with the --force option, and edit the target branch of this pull request.

@dehilsterlexis

**Comment by dehilsterlexis** [comment]
@richardkchapman back to you

**Comment by richardkchapman** [comment]
@GordonSmith Ok to merge?

File: comms/ModFileAttribute.cpp [hunk 4]
```diff
@@ -70,6 +70,11 @@ class CModFileAttribute : public IAttribute, public IAttributeHistory, public cl
     {
     }
 
+    int PreProcess(PREPROCESS_TYPE action, const TCHAR * overrideEcl, IAttributeVector & attrs, IAttributeBookkeep & attrProcessed, Dali::CEclExceptionVector & errs) const
+    {
+        return 0;
+    }
+
```
Comment (GordonSmith) [inline comms/ModFileAttribute.cpp]: This file already has a "PreProcess" method whose signature needs to change to match this one.

File: comms/ModFileRepository.cpp [hunk 5]
```diff
@@ -238,7 +238,7 @@ class CModFileRepository : public IRepository, public clib::CLockableUnknown
             boost::algorithm::trim(trimmedLine);
             if (boost::algorithm::istarts_with(trimmedLine, IMPORT_MARKER))
             {
-                if (attributeLabel.length() && attributeEcl.length())
+                if (attributeLabel.length())
```
Comment (GordonSmith) [inline comms/ModFileRepository.cpp]: attributeEcl always has content now?  (defaults to ecl?)

**Comment by richardkchapman** [comment]
@GordonSmith Back to you?

**Comment by GordonSmith** [comment]
@dehilsterlexis needs rebase.

**Pull request merged**